ckanext/recombinant/datatypes.py

```diff
@@ -117,7 +122,7 @@ def canonicalize(
     # leads to unpleasantness
     if primary_key:
         dirty = dirty.strip()
-        dirty = re.sub(u'[\x00-\x1f]', '', dirty)
+        dirty = re.sub(r'[\x00-\x1f]', '', dirty)
```

Comment:
this changes the interpretation of the literal string, does a `\x##` work in regexes?

Reply:
hmmm I don't think this would work, so will switch back to normal string